src/storage.zig

```diff
@@ -7,400 +7,555 @@ const log = std.log.scoped(.storage);
-                    // also have experienced the disk's read retry timeout (in seconds).
-                    // TODO Our docs must instruct on why and how to reduce disk firmware timeouts.
-
-                    // These lines both implement ceiling division e.g. `((3 - 1) / 2) + 1 == 2` and
-                    // require that the numerator is always greater than zero:
-                    assert(target.len > 0);
-                    const target_sectors = @divFloor(target.len - 1, constants.sector_size) + 1;
-                    assert(target_sectors > 0);
-                    read.target_max =
-                        (@divFloor(target_sectors - 1, 2) + 1) * constants.sector_size;
-                    assert(read.target_max >= constants.sector_size);
-
-                    // Pass 0 for `bytes_read`, we want to retry the read with smaller `target_max`:
-                    self.start_read(read, 0);
-                    return;
-                } else {
-                    // We tried to read at (or less than) logical sector granularity and failed.
-                    log.err("latent sector error: offset={}, zeroing sector...", .{read.offset});
-
-                    // Zero this logical sector which can't be read:
-                    // We will treat these EIO errors the same as a checksum failure.
-                    // TODO This could be an interesting avenue to explore further, whether
-                    // temporary or permanent EIO errors should be conflated with checksum failures.
-                    assert(target.len > 0);
-                    @memset(target, 0);
-
-                    // We could set `read.target_max` to `vsr.sector_ceil(read.buffer.len)` here
-                    // in order to restart our pseudo-binary search on the rest of the sectors to be
-                    // read, optimistically assuming that this is the last failing sector.
-                    // However, data corruption that causes EIO errors often has spacial locality.
-                    // Therefore, restarting our pseudo-binary search here might give us abysmal
-                    // performance in the (not uncommon) case of many successive failing sectors.
-                    self.start_read(read, target.len);
-                    return;
-                }
-            },
-
-            error.WouldBlock,
-            error.NotOpenForReading,
-            error.ConnectionResetByPeer,
-            error.Alignment,
-            error.IsDir,
-            error.SystemResources,
-            error.Unseekable,
-            error.ConnectionTimedOut,
-            error.Unexpected,
-            => {
-                log.err(
-                    "impossible read: offset={} buffer.len={} error={s}",
-                    .{ read.offset, read.buffer.len, @errorName(err) },
-                );
-                @panic("impossible read");
-            },
-        };
+        const write_buf = written_storage_data[zone.start() + offset_in_zone ..][0..len];
```

Comment:
[Try to avoid abbreviations](https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/TIGER_STYLE.md#developer-experience) where possible, so `write_buffer` and `length`.